docs/local_assembly.tex

```diff
@@ -58,12 +58,19 @@ \section{Correcting Reads} \label{correcting-reads}
+Before we discuss how to construct the graph, let us confess a slight abuse of terminology.  We refer to our assembly graph as a de Bruijn graph, that is, a graph in which the vertices are kmers and two kmers have a directed edge whenever they appear consecutively in a read.  This is almost true (until the last part of assembly when we convert the graph into a more compact sequence graph), but there are a few differences:
 
-The first step is to create a set of sequences to be kmerized and put into the de Bruijn graph.  These sequences are: the reference haplotype, any requested alleles given by the \code{alleles} argument in \HC's ``genotype given alleles" (GGA) mode\footnote{This mode is disabled in \Mutect~ and incompatible with the reference confidence mode of \HC, which is the recommended best practice.  We document it here for completeness.}, and maximal subsequences of reads with base quality at least 10 by default\footnote{This is controlled by the \code{minBaseQualityScore} argument.} and at least one kmer long.  That is, a read with 100 bases and a base of quality 7 at position 70 yields sequences of length 69, 30 if $k = 10$ and a single sequence of length 69 if $k = 40$.  We do not use mate information; that is, a read and its mate yield completely independent sequences, nor do we later post-process the graph using mate information.  This is a potential direction for improvement.
+\begin{itemize}
+\item The edges of our graph record their multiplicity, the number of times the kmers they connect occur consecutively.
+\item The kmers of our graph come not just from reads but also the reference haplotype and, in GGA mode (see below), haplotypes generates from the given alleles.
+\item A non-unique kmer (see below) may be associated with multiple vertices.
+\end{itemize}
+
+Next, the \code{ReadThreadingAssembler} assembles the (corrected) reads over several different kmer sizes specified by the \code{kmerSize} argument\footnote{By default, 10 and 25.  This size is a compromise and no single value is the best choice for all regions.  Large kmers are more likely to be unique and to yield a graph with no cycles, which is especially important in low-complexity regions, but they are more sensitive to errors and low coverage.}.  If the given kmer sizes fail to produce a graph without cycles, or if more than $1/5$ of the kmers in the graph are not unique in the sequences from which they come\footnote{This uniqueness condition is waived on the last attempt when $k$ has been increased by 60 bases.}, the \code{ReadThreadingAssembler} repeatedly increases $k$ by 10 bases, starting from the largest initial $k$, until assembly succeeds or until $k$ has been increased 6 times, that is, by 60 bases\footnote{These constants, 10 and 6, are hard-coded, but the attempts to increase $k$ can be disabled completely with the \code{dontIncreaseKmerSizesForCycles} argument.}.
```

Comment:
> Large kmers are more likely to be unique and to yield...
> but they are more sensitive to errors and low coverage.

mmm... why you're saying that? has it been shown to be the case.

I think larger could would be less sensitive sometimes
for example they may fail to close back into the reference after a even close to the end of the active region ... should be recoverable as a dangling tail but not always. The viceversa could be true with dangling heads.

I don't think that anyone has ever do a proper analysis of this so is all conjecture... but if you are confident that what you are saying is accurate then go ahead.

Reply:
I think I was saying what you're saying but with a confusing overload of the word "sensitive."  I meant that large kmers would be more *affected* by errors, which could decrease sensitivity.  I'll make that more clear.